src/lib.rs

```diff
@@ -459,7 +459,25 @@ impl Build {
             shell_escaped_flags: None,
             build_cache: Arc::default(),
             inherit_rustflags: true,
+        };
+        if cfg!(target_os = "solaris") || cfg!(target_os = "illumos") {
```

Comment:
It's better to put this in `create_compile_object_cmd`

A simple `cfg!` won't work for cross compilation, so we would need the target.

In that function there's a `target` variable that contains parsed target information that you could use to determine the os

Reply:
ah that makes sense, sorry about the oversight. I moved it to `add_default_flags` so that `CRATE_CC_NO_DEFAULTS` suppresses it like others. `try_get_compiler` uses that, and is also used for `expand`/`try_expand` that ought to get expanded under the same conditions as compilation too.

How's this look?